pkgs/kde/third-party/wallpaper-engine-plugin/default.nix

```diff
@@ -0,0 +1,60 @@
+{
+  extra-cmake-modules,
+  fetchFromGitHub,
+  full,
+  kpackage,
+  libplasma,
+  lib,
+  lz4,
+  mkKdeDerivation,
+  mpv,
+  pkg-config,
+  python3,
+}:
+mkKdeDerivation {
+  pname = "wallpaper-engine-kde-plugin";
+  version = "qt6";
+
+  src = fetchFromGitHub {
+    owner = "catsout";
+    repo = "wallpaper-engine-kde-plugin";
+    rev = "1e604105c586c7938c5b2c19e3dc8677b2eb4bb4";
+    hash = "sha256-bKGQxyS8gUi+37lODLVHphMoQwLKZt/hpSjR5MN+5GA=";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./nix-plugin.patch ];
+
+  extraBuildInputs = [
+    extra-cmake-modules
+    full
+    libplasma
+    lz4
+    mpv
+  ];
+
+  extraCmakeFlags = [ "-DUSE_PLASMAPKG=ON" ];
+
+  extraNativeBuildInputs = [
+    kpackage
+    pkg-config
+  ];
+
+  postInstall =
+    let
+      py3-ws = python3.withPackages (ps: with ps; [ websockets ]);
+    in
+    ''
+      cd ../plugin
+      PATH=${py3-ws}/bin:$PATH patchShebangs --build ./contents/pyext.py
```

Comment:
Can't we put this into nativeBuildInputs instead?

Reply:
Could be propagated from somewhere, I couldn't find it in the mkKdeDerivation wrapper.

Reply:
I think that's a separate issue that should probably be addressed in another PR. I also notice similar workaround in other packages as well, so I think this might be fine for now.